src/number_theory/number_field/embeddings.lean

```diff
@@ -338,10 +338,18 @@ begin
   { exact λ h, ⟨embedding w, h, mk_embedding w⟩, },
 end
 
-lemma not_is_real_iff_is_complex {w : infinite_place K} :
-  ¬ is_real w ↔ is_complex w :=
+@[simp] lemma not_is_real_iff_is_complex {w : infinite_place K} : ¬ is_real w ↔ is_complex w :=
 by rw [is_complex_iff, is_real_iff]
 
+@[simp] lemma not_is_complex_iff_is_real {w : infinite_place K} : ¬ is_complex w ↔ is_real w :=
+by rw [←not_is_real_iff_is_complex, not_not]
+
+lemma is_real_or_is_complex (w : infinite_place K) : is_real w ∨ is_complex w :=
+by { rw ←not_is_real_iff_is_complex, exact em _ }
+
+lemma is_complex_or_is_real (w : infinite_place K) : is_complex w ∨ is_real w :=
```

Comment:
Is this lemma really needed? It seems a bit random to have the sides swapped of some or statement as a separate lemma

Reply:
Pinging @YaelDillies that added these lemmas

Reply:
This matches `em` and `em'` and helps deal with the easy case of the proof first when doing cases on such an hypothesis, the point being that the easy case isn't the same branch in all lemmas.

Reply:
Feel free to drop it, but I see no reason to prefer `is_real` over `is_complex`, so I would rather have both.

Reply:
Well in number theory we almost always put the real embeddings "on the left" in some sense so that's a reason.
Really though you could say the same about the two sides of very many or statements right, if there is indeed symmetry in the setup there is often no reason to prefere one side, so I'm in favour of only keeping version of such lemmas and getting the rest with symm.